### DISCUSSION TITLE
Switch DirichletBC to NeumannBC on the same boundary

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#17238] [Q&A General] [answered]
Dear MOOSE Team,

I need to set DirichletBC at time = 0, then switch the DirichletBC to NeumannBC for time > 0 on the same boundary.
Please help.

S. Thomas

**Comment by josebastiase** [comment]
Hi,
I'm not sure whether this is implemented in MOOSE already, but I would try using controls.
https://mooseframework.inl.gov/syntax/Controls/
Cheers,
Jose

**Comment by ngrilli** [comment, marked as answer]
Dear @styyokuda 

If it's just at time 0, you could set an initial condition on the variable of interest using FunctionIC.
By setting the initial condition as a function of the coordinates x,y,z
you can make sure the initial condition is applied only on the boundary of choice at time 0.

Otherwise, what I did in the past was to use the "basic restart" procedure described here:
https://mooseframework.inl.gov/modules/porous_flow/restart.html

In the restart input file you can just remove the DirichletBC and the solver will apply NeumannBC
in the restarted simulation.

Trust this helps.

Best Regards,
Nicolò Grilli
National University of Singapore
